src/planner/binder/expression/bind_columnref_expression.cpp

```diff
@@ -222,20 +217,29 @@ unique_ptr<ParsedExpression> ExpressionBinder::CreateStructPack(ColumnRefExpress
 }
 
 unique_ptr<ParsedExpression> ExpressionBinder::QualifyColumnName(ColumnRefExpression &colref, string &error_message) {
+
+	// try binding as a lambda parameter
+	auto lambda_ref = LambdaRefExpression::FindMatchingBinding(lambda_bindings, colref.ToString());
```

Comment:
Calling `ToString` seems somewhat error prone here w.r.t. escaping. What if we have a lambda called `x.y` and we then use the qualified column reference `x.y`? For example:

```sql
select list_transform([1,2,3], "x.y" -> "x.y" + x.y) as l from (values (42), (84)) x(y);

```